### PR TITLE
Fix lsp status messages

### PR DESCRIPTION
**Pull request by kuznetsss**
I've fixed errors and add some improvements. Now it works fine.

But I have a question: is it possible to prevent flickering on typing (see the video)?
Clangd sends messages `file is queued` and then `idle` for any changes in buffer so it's a little annoying.
I tried to fix it by saving last showing time but it breaks auto update of message for some reason. Sometimes it doesn't update util I move cursor. And lua can't count time more accurate then 1 second. Do you have any ideas?

https://user-images.githubusercontent.com/15742918/108608012-fd9b3380-73d4-11eb-9547-bbe3c0dfa632.mp4

### CODE REVIEW

**Comment by datwaft** [comment]
Maybe using vim built-in functions to keep track of time, like [nvim-lua-guide](https://github.com/nanotee/nvim-lua-guide#the-vim-namespace) says.

**Comment by kuznetsss** [comment]
Thank you. I've found out how to fix that.

Could you also add delay time to options?
It's `500`in the [line](https://github.com/datwaft/bubbly.nvim/pull/87/files#diff-34ba23dd08648304eb491ee76927e1dfe113beda382ebc8eca214e57f4cbf88fR84).

**Comment by datwaft** [comment]
> Could you also add delay time to options?

Sure! Maybe a variable called `g:bubbly_timing` that follows the structure:

```lua
vim.g.bubbly_timing = {
   lsp_status = {
      messages = 500,
   },
}
```
What do you think?

**Comment by kuznetsss** [comment]
> What do you think?

The structure is good, but maybe use more specific name `messages_update_delay` instead of just `messages`?
I think that would be a little more clear what is this option about.

```lua
vim.g.bubbly_timing = {
   lsp_status = {
      messages_update_delay = 500, -- ms
   },
}
```

**Comment by datwaft** [comment]
What do you think about this?
```lua
vim.g.bubbly_timing = {
   lsp_status = {
      messages = {
         update_delay = 500, -- ms
      }
   },
}
```
I prefer to have it separated, for if some change requires that messages has multiple delays.

**Comment by kuznetsss** [comment]
> What do you think about this?

This is even better.

**Review by datwaft — APPROVED** [review]

**Comment by datwaft** [comment]
Do I merge this now and add the configuration option? Or do you have any additional change you want to make?

File: lua/bubbly/components/lsp_status/messages.lua [hunk 1]
```diff
@@ -17,7 +17,7 @@ if not settings.style then
    settings.style = vim.g.bubbly_styles.default
 end
 
-local get_messages = require'bubbly.utils.prerequire''lsp-status.messaging'
+local get_messages = require'bubbly.utils.prerequire''lsp-status/messaging'
```
[the remark this answers is not in the document]
Comment (kuznetsss) [reply lua/bubbly/components/lsp_status/messages.lua]: Thank you for mention that. I've fixed wrong require.

**Comment by kuznetsss** [comment]
> Do I merge this now and add the configuration option? Or do you have any additional change you want to make?

It works fine now so you could merge it.

**Comment by kuznetsss** [comment]
Now definitely everything is correct:smiley:

**Review by datwaft — APPROVED** [review]

**Comment by datwaft** [comment]
Merged!

**Pull request merged**